docs/specs/config.md

```diff
@@ -0,0 +1,273 @@
+
+### `dotnet/docs`: Complicate Metadata
+https://github.com/dotnet/docs/blob/master/docfx.json#L84-L126
+``` yml
+metadata:
+- path: "**"
+  values:
+    breadcrumb_path: /dotnet/breadcrumb/toc.json
+    _displayLangs: ["csharp"]
+    author: dotnet-bot
+    ms.author: dotnetcontent
+    manager: wpickett
+    searchScope: [".NET"],
+    uhfHeaderId: MSDocsHeader-DotNet
+    apiPlatform: dotnet
+- path: _csharplang/spec
+  values:
+    ms.prod: .net
+    ms.topic: language-reference
+    ms.date: 07/01/2017
+    ms.technology: devlang-csharp
+    ms.author: wiwagn
+- path: _vblang/spec
+  values:
+    ms.prod: .net
+    ms.topic: language-reference
+    ms.date: 07/21/2017
+    ms.technology: devlang-visual-basic
+    ms.author: wiwagn
+- path: csharp/quick-starts
+  values:
+    ms.technology: csharp-interactive
+- path:
+  - docs/core
+  - docs/csharp
+  - docs/framework
+  - docs/fsharp
+  - docs/standard
+  - docs/visual-basic
+  values:
+    dev_langs: vb
+```
+Global/File metadata is merged into single `metadata`:
+| Key           | Optional? | Type         | Description |
+|:-------------:|:---------:|:------------:|-------------|
+| paths         | Y         | string/array | one/multiple filename/foldername(s). Omit for global metadata |
+| values        |           |   object | the key-value pair of metadata applied to paths |
+
+### `ATADocs-pr`: Multiple Docset
+[op.config](https://github.com/MicrosoftDocs/ATADocs-pr/blob/master/.openpublishing.publish.config.json)
+``` yml
+name: Azure.ATADocs
+include: "{ATADocs,ATPDocs}/**/*.{md,svg,png,jpg,jpeg,gif,svg}"
+metadata:
+- path: "**"
+  values:
+    layout: Conceptual
+    breadcrumb_path: /enterprise-mobility-security/toc.json
+- path: ATPDocs/**
+  values:
+    extendBreadcrumb: true
+routes:
+  ATADocs/DeployUse/**: advanced-threat-analytics/deploy-use
+  ATADocs/**: advanced-threat-analytics
+  ATADocs/PlanDesign/**: advanced-threat-analytics/plan-design
+  ATADocs/Troubleshoot/**: advanced-threat-analytics/troubleshoot
+  ATADocs/Understand/**: advanced-threat-analytics/understand-explore
+  ATPDocs/**: azure-advanced-threat-protection
+```
+No config is set to docset level.
+* `name`: docset's name is preserved for generating documentId. We assume multiple docsets share the same name in a repo. Otherwise, the config is a bit complicated like below:
+  ``` yml
+  name:
+    ATADocs/DeployUse: ATADeployUse
+    ATADocs/PlanDesign: ATAPlanDesign
+    ATADocs/Troubleshoot: ATATroubleshoot
+    ATADocs/Understand: ATAUnderstand
+    ATADocs: ATADocs
+    ATPDocs: ATPDocs
+  ```
+* `basePath`: It is used to calculate canonical URL. Not needed here as the basePath is included in `routes` now.
+* `product`: Merged into name.
+
+### `sql-docs-pr`: Complicate Versioning
+[docfx.json](https://github.com/MicrosoftDocs/sql-docs-pr/blob/release-ops-versioning-2/docs/docfx.json)
+``` yml
+name: SQL.sql-content
+include: "docs/**/*.{md,yml}"
+exclude: "docs/mref/**"
+monikerRange:
```

Comment:
>monikerRange [](start = 0, length = 12)

why it is not a array? and what's about the two folders have the same moniker? and what's the matching order?